### DISCUSSION TITLE
"Enable Airplay mode (experimental)" disabled for an S2 sonos amp

### DISCUSSION THREAD

**Discussion by smshapiro85** [music-assistant/support#3282] [Q&A] [answered]
Hello community! I have a few sonos amps on S2, and I have added them to music-assistant and can play music to them successfully, but the button to "Enable Airplay mode (experimental)" is disabled. I have searched through this discussion board, google and chatgpt with no success, so I wanted to ask the community for more information.

On HAOS with home assistant core v. 2024.12.1, and the sonos amps are on Sonos OS S2, version 16.4.4.

The Airplay provider is enabled and working.

Screenshot of sonos speaker config:

<img width="1312" alt="image" src="https://github.com/user-attachments/assets/afe1aaf3-25d3-44e5-a201-de62c430fbb3">

Settings config:

<img width="1229" alt="image" src="https://github.com/user-attachments/assets/1b34fdf9-1db2-4448-b3a9-a08604064169">

Thank you for your help!

**Comment by OzGav** [comment, marked as answer]
Newer generation Sonos speakers appear to no longer support Airplay 1 therefore using the Airplay provider to stream to these speakers is not possible.

**Reply by smshapiro85** [reply]
Ah, I see, so the only way to play at the same time is to create a universal speaker group? Thank you for your help!

**Reply by OzGav** [reply]
Only if you are trying to sync both series 1 and 2 speakers together. Otherwise you can sync S2 speakers together or S1 speakers together

**Reply by smshapiro85** [reply]
Sorry to clarify- an AirPlay speaker with an S2 speaker.
Once again thank you very much for your help!

**Reply by OzGav** [reply]
Thanks. To summarise then, If the AirPlay option is greyed out on the a Sonos player it is not supported for that player. Yes you can use a universal group but the players won’t be in sync,

**Reply by smshapiro85** [reply]
Thank you Oz! One last question- have you heard anything on Music Assistant potentially supporting Airplay 2 in the future? I currently airplay to my sonos amps currently and it works flawlessly, so maybe one day this could be a possibility. Thank you again for your help.

**Reply by OzGav** [reply]
Airplay 2 has not been reverse engineered so whilst it is technically possible there is no time for it at the moment.

**Reply by ajweber** [reply]
@OzGav https://github.com/JohannVR/JohannVRs-Home-Assistant-Addons/tree/main/Airplay2 ?

**Reply by OzGav** [reply]
@ajweber no that allows a device to function as an AirPlay 2 RECEIVER. We need to be able to TRANSMIT

**Reply by ajweber** [reply]
Ah, I am new to MA...actually just getting ready to install it.  Does HM already allow to be an AirPlay RECEIVER?  That is, can my kids find one of my configured players with their iOS device (Sonos S1, probably) and stream music?

Sorry for the newbie interruption. :)

**Reply by marcelveldt** [reply]
Not yet, but that feature is currently in the making; so you can make any MA player an airplay and/or spotify connect target

**Comment by marcelveldt** [comment]
The Sonos AMP should still work - the only devices I'm aware of that have a broken Airplay RAOP implementation are ERA 100, ERA 300, Move 2 and Arc Ultra. Did you enable the Airplay variant of these speakers ?

**Reply by ajweber** [reply]
Honestly don't know what an "Airplay variant" of a Sonos AMP is.  Are you referring to configuration inside MA?

**Reply by OzGav** [reply]
I think Marcel meant did you enable the AirPlay option for that Sonos player?

**Reply by marcelveldt** [reply]
Yes, enable the Airplay provider in Music Assistant. That should also auto-detect the Airplay variants of your players. Now go into the details of those discovered players and mark them as "hidden in UI" to prevent you from seeing each speaker twice.

The feature basically just forwards all the playback related commands to that airplay player under the hood.

**Reply by ajweber** [reply]
OK, sorry for being a pest, but will I still be able to use the direct, Sonos S1 (Android) app to control my amps when MA is not streaming audio to them?  Am I switching the Sonos devices/players to an Airplay "mode", or am I just enabling the Airplay support as an alternative source?

**Reply by OzGav** [reply]
This just affects what MA does
